clients/include/utility.h

```diff
@@ -525,63 +530,97 @@ class Arguments
     rocblas_int norm_check = 0;
     rocblas_int unit_check = 1;
     rocblas_int timing     = 0;
-
-    rocblas_int iters = 10;
+    rocblas_int iters      = 10;
 
     uint32_t algo          = 0;
     int32_t solution_index = 0;
     uint32_t flags         = 0;
     size_t workspace_size  = 0;
 
-    Arguments& operator=(const Arguments& rhs)
+    char function[32] = "";
+    char namex[32]    = "";
+    char category[32] = "";
+
+    // Function to read Structures data from stream
+    friend istream& operator>>(istream& s, Arguments& arg)
+    {
+        s.read(reinterpret_cast<char*>(&arg), sizeof(arg));
+        return s;
+    }
+
+    // Function to print Structures data out to stream (for debugging)
+    friend ostream& operator<<(ostream& o, const Arguments& arg)
+    {
+        return o << "{ 'transA': '" << arg.transA_option << "', 'transB': '" << arg.transB_option
+                 << "', 'M': '" << arg.M << "', 'N': '" << arg.N << "', 'K': '" << arg.K
+                 << "', 'lda': '" << arg.lda << "', 'ldb': '" << arg.ldb << "', 'ldc': '" << arg.ldc
+                 << "', 'alpha': " << arg.alpha << ", 'beta': " << arg.beta << " }\n";
+    }
+};
+
+enum rocblas_data_class
+{
+    rocblas_test_data,
+    rocblas_perf_data,
+};
+
+// Class used to read Arguments data into the tests
+template <rocblas_data_class>
+struct RocBLAS_Data
```

Comment:
This is the Data file iterator class. It contains a singleton which is initialized with a filename on the first call.

The data file is read an arbitrary number of times, by calling the begin() method and iterating until it reaches end(), then calling begin() again, it resets to the beginning of the file. This allows an arbitrary number of passes through the data file. The begin() method allows an option filter functor which returns true when a datum is to be admissible.